sentry-spring/src/main/java/io/sentry/spring/tracing/SentryTracingFilter.java

```diff
@@ -85,7 +87,7 @@ protected void doFilterInternal(
         // if transaction name is not resolved, the request has not been processed by a controller
         // and we should not report it to Sentry
         if (transactionName != null) {
-          transaction.setName(transactionName);
+          transaction.setName(transactionName, TransactionNameSource.ROUTE);
```

Comment:
Better to check with other folks, shouldn't that be `URL`?

Reply:
Hmm you're right, it's only `ROUTE` for `SpringMvcTransactionNameProvider` but since people might provide their own, it should probably rather be `CUSTOM` for now. Not sure if we need to improve the `TransactionNameProvider` interface and add a method with default impl that returns `CUSTOM` marked as internal which we override and provide what we think works. Not sure I like this approach though.

Reply:
Created https://github.com/getsentry/sentry-java/pull/2188 to show what I mean. Not sure I like the approach though. As an alternative easy way out we could simply set `CUSTOM` as `source` instead.

Reply:
Merged them in here now as it's still better than sending wrong `source`